packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -31,39 +39,76 @@ export abstract class ForceSourceDeployExecutor extends SfdxCommandletExecutor<
 
-    let stdErr = '';
-    execution.stderrSubject.subscribe(realData => {
-      stdErr += realData.toString();
+    channelService.streamCommandStartStop(execution);
+    channelService.showChannelOutput();
+
+    let stdOut = '';
+    execution.stdoutSubject.subscribe(realData => {
+      stdOut += realData.toString();
     });
 
     execution.processExitSubject.subscribe(async exitCode => {
       this.logMetric(execution.command.logName, startTime);
-      if (exitCode !== 0) {
-        try {
-          const deployErrorParser = new ForceDeployErrorParser();
-          const fileErrors = deployErrorParser.parse(stdErr);
+      try {
+        const deployParser = new ForceDeployResultParser(stdOut);
+        const errors = deployParser.getErrors();
+        if (errors) {
           handleDiagnosticErrors(
-            fileErrors,
+            errors,
             workspacePath,
             execFilePathOrPaths,
             ForceSourceDeployExecutor.errorCollection
           );
-        } catch (e) {
-          telemetryService.sendError(
-            'Error while creating diagnostics for vscode problem view.'
-          );
-          console.error(
-            'Error while creating diagnostics for vscode problem view.'
-          );
+        } else {
+          ForceSourceDeployExecutor.errorCollection.clear();
         }
-      } else {
-        ForceSourceDeployExecutor.errorCollection.clear();
+        this.outputResult(deployParser);
+      } catch (e) {
+        telemetryService.sendError(
+          'Error while creating diagnostics for vscode problem view.'
+        );
+        console.error(
+          'Error while creating diagnostics for vscode problem view.'
+        );
       }
     });
 
-    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  private outputResult(parser: ForceDeployResultParser) {
+    const table = new Table();
+    const errors = parser.getErrors();
+    const successes = parser.getSuccesses();
+    const deployedSource = successes ? successes.result.deployedSource : undefined;
+
+    if (deployedSource) {
+      const outputTable = table.createTable(deployedSource as unknown as Row[], [
+        { key: 'state', label: nls.localize('table_header_state') },
+        { key: 'fullName', label: nls.localize('table_header_full_name') },
+        { key: 'type', label: nls.localize('table_header_type') },
+        { key: 'filePath', label: nls.localize('table_header_project_path') }
+      ]);
+      channelService.appendLine('=== Deployed Source');
+      channelService.appendLine(outputTable);
+    }
+
+    if (errors) {
+      const outputTable = table.createTable(errors.result as unknown as Row[], [
```

Comment:
When running a deploy from a `package.xml` and it has deploy errors, all the error parsing seems correct up until this point since the `errors` object does not have a `result` property and causes the rendering of the table to fail.

![screen shot 2019-02-26 at 9 26 20 am](https://user-images.githubusercontent.com/1041842/53433434-37cdc800-39a9-11e9-9594-82f374dfcb32.png)